examples/amqp/main.go

```diff
@@ -155,15 +155,24 @@ func main() {
 }
 
 func getAWSSession(endpoint string) *session.Session {
-	return session.Must(
-		session.NewSession(
+
```

Comment:
I have enhanced this function with reattempts because otherwise, in the e2e flow the service would frequently fail to start due to being ready before the `localstack` docker container. With this enhancement The service has a small re-attempt logic embed which in my case represented 100%  of success rate of the e2e flow. The 100% measurement is out of a sample of 15 runs.

Reply:
Will definitely check it out and come back to you.

Reply:
Dear @mantzas,

After checking the code in `aws_test.go` here is what I understood about it, please correct me if I misunderstood your point:

The 'retry' method on `aws_test.go` uses an exponential backoff mechanism baked on the library __dockertest__ to access the runtime of the dependant container. Sounds like the perfect solution in the context of the integration tests.

On the other hand, in the example of the `examples` e2e tests, we have no access to the underlying runtime and it's not within the responsability of the clients (e.g. `amqp/main.go`) to have knowledge on which runtime environment the 'infrastructure' is running on (in this case infrastructure == 'localstack container'). 
What I tried to search and was not able is a built-in session-acquisition API in the `aws-sdk-go` lib. 

p.s.: this comment made me get back to my code and realize that I was missing the 'time.Sleep' call in the retry logic :yum: